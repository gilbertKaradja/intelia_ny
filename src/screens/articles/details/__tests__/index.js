import React from 'react';
import { shallow, mount } from 'enzyme';

import DetailsScreen from '../index.js';

let sampleData = {
    "url": "https://www.nytimes.com/interactive/2019/02/15/upshot/british-irish-dialect-quiz.html",

    "section": "The Upshot",
    "byline": "By JOSH KATZ",

    "title": "The British-Irish Dialect Quiz",
    "abstract": "What does the way you speak say about where you’re from? Answer 25 questions to see your own custom dialect map.",
    "published_date": "2019-02-15",
    "source": "The New York Times",
    "id": 100000006354694,

    "media": [
        {
            "media-metadata": [
                {
                    "url": "https://static01.nyt.com/images/2019/02/14/us/british-irish-dialect-quiz-promo-1550183830563/british-irish-dialect-quiz-promo-1550183830563-thumbStandard-v3.jpg",
                    "format": "Standard Thumbnail",
                    "height": 75,
                    "width": 75
                },
                {
                    "url": "https://static01.nyt.com/images/2019/02/14/us/british-irish-dialect-quiz-promo-1550183830563/british-irish-dialect-quiz-promo-1550183830563-mediumThreeByTwo440-v3.jpg",
                    "format": "mediumThreeByTwo440",
                    "height": 293,
                    "width": 440
                }
            ]
        }
    ]
};

let sampleData2 = {
    "url": "https://www.nytimes.com/interactive/2019/02/15/upshot/british-irish-dialect-quiz.html",

    "section": "The Upshot",
    "byline": "By JOSH KATZ",

    "title": "The British-Irish Dialect Quiz",
    "abstract": "What does the way you speak say about where you’re from? Answer 25 questions to see your own custom dialect map.",
    "published_date": "2019-02-15",
    "source": "The New York Times",
    "id": 100000006354694,

    "media": ''
};



describe('DetailsScreen component', () => {

    it('should render without issues', () => {

        //renders initial dropdown properly
        const component = shallow(
            <DetailsScreen
                articleData={sampleData}
            />
        );
        expect(component).toMatchSnapshot();

        component.unmount();
    });


    it('should render properly even if an image is not available', () => {

        //renders initial dropdown properly
        const component = shallow(
            <DetailsScreen
                articleData={sampleData2}
            />
        );
        expect(component).toMatchSnapshot();

        component.unmount();
    });



    it('should call the back handler function when back button is clicked/pressed', () => {

        const spy = jest.spyOn(DetailsScreen.prototype, 'goBack');

        const component = mount(
            <DetailsScreen
                articleData={sampleData}
                history={{goBack: () => {}}}
            />
        );

        component.instance().goBack();

        expect(spy).toHaveBeenCalled();

        spy.mockClear()
    });

    it('should render nothing on empty props and call the back handler', () => {

        const spy = jest.spyOn(DetailsScreen.prototype, 'goBack');

        const component = mount(
            <DetailsScreen
                articleData={null}
                history={{goBack: () => {}}}
            />
        );

        expect(component).toMatchSnapshot();

        component.instance().goBack();

        expect(spy).toHaveBeenCalled();

        spy.mockClear()
    });
});