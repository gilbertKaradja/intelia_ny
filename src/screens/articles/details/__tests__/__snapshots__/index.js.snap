// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsScreen component should render properly even if an image is not available 1`] = `
<div
  className="DetailsScreen"
>
  <div
    className="header"
  >
    <div
      className="wrapper"
    >
      <div
        className="button_container"
        onClick={[Function]}
      >
        <ReactComponent />
      </div>
      <div
        className="text"
      >
        <div
          className="title"
        >
          The New York Times
        </div>
        <div
          className="section"
        >
          The Upshot
        </div>
      </div>
      <div
        className=""
      />
    </div>
  </div>
  <div
    className="content"
  >
    <div
      className="wrapper"
    >
      <div
        className="image_wrapper"
      >
        <FadeImage
          src=""
        />
      </div>
      <div
        className="article_details"
      >
        <div
          className="title"
        >
          The British-Irish Dialect Quiz
        </div>
        <div
          className="author"
        >
          By JOSH KATZ
        </div>
        <div
          className="date"
        >
          2019-02-15
        </div>
        <div
          className="abstract"
        >
          What does the way you speak say about where you’re from? Answer 25 questions to see your own custom dialect map.
        </div>
        <div
          className="redirect_wrapper"
        >
          <a
            href="https://www.nytimes.com/interactive/2019/02/15/upshot/british-irish-dialect-quiz.html"
            rel="noopener noreferrer"
            target="_blank"
          >
            Read More
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DetailsScreen component should render without issues 1`] = `
<div
  className="DetailsScreen"
>
  <div
    className="header"
  >
    <div
      className="wrapper"
    >
      <div
        className="button_container"
        onClick={[Function]}
      >
        <ReactComponent />
      </div>
      <div
        className="text"
      >
        <div
          className="title"
        >
          The New York Times
        </div>
        <div
          className="section"
        >
          The Upshot
        </div>
      </div>
      <div
        className=""
      />
    </div>
  </div>
  <div
    className="content"
  >
    <div
      className="wrapper"
    >
      <div
        className="image_wrapper"
      >
        <FadeImage
          src="https://static01.nyt.com/images/2019/02/14/us/british-irish-dialect-quiz-promo-1550183830563/british-irish-dialect-quiz-promo-1550183830563-mediumThreeByTwo440-v3.jpg"
        />
      </div>
      <div
        className="article_details"
      >
        <div
          className="title"
        >
          The British-Irish Dialect Quiz
        </div>
        <div
          className="author"
        >
          By JOSH KATZ
        </div>
        <div
          className="date"
        >
          2019-02-15
        </div>
        <div
          className="abstract"
        >
          What does the way you speak say about where you’re from? Answer 25 questions to see your own custom dialect map.
        </div>
        <div
          className="redirect_wrapper"
        >
          <a
            href="https://www.nytimes.com/interactive/2019/02/15/upshot/british-irish-dialect-quiz.html"
            rel="noopener noreferrer"
            target="_blank"
          >
            Read More
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
